### PR TITLE
Bobbrow/webpack

### PR DESCRIPTION
**Pull request by sean-mcmanus**
Debugging doesn't work. Something wrong with the source mapping I guess.

### CODE REVIEW

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -336,6 +339,7 @@ class DefaultClient implements Client {
                     // Listen for messages from the language server.
                     this.registerNotifications();
                     this.registerFileWatcher();
+                    console.log('registered stuff');
```
Comment (bobbrow) [inline Extension/src/LanguageServer/client.ts]: probably don't want to keep these logs. :)

Comment (sean-mcmanus) [reply Extension/src/LanguageServer/client.ts]: Oh, I thought those logs were intentional. Although "registered stuff" seemed odd.

**Comment by sean-mcmanus** [comment]
Looks like this breaks the Azure Pipeline build...even though building works locally. Error: Cannot find module './out/tools/GenerateOptionsSchema'

**Review by WardenGnaw — COMMENTED** [review]
I wouldn't webpack src/tools, since it is not used for the released VsCode extension.

I would add another agent build for testing webpacked builds and keep our normal just tsc builds since we do not want webpacking for regular dev workflow. 

Add Extension/dist/* to .gitignore

**Comment by bobbrow** [comment]
When the Config UI goes in, we'll need to update this to make sure all the assets are in the right place.

File: Extension/tools/prepublish.js [hunk 2]
```diff
@@ -10,23 +10,36 @@
...
 }
 
-// Compile the TypeScript code
-console.log(">> tsc -p ./");
-cp.execSync("tsc -p ./", {stdio:[0, 1, 2]});
+const tscCompileListStr = fs.readFileSync("./tscCompileList.txt").toString();
+
+tscCompileListStr.split(os.EOL).forEach(filePath => {
+    if (!filePath.startsWith("#") && /\S/.test(filePath))
+    {
```
Comment (bobbrow) [inline Extension/tools/prepublish.js]: move curly brace to previous line

File: Extension/tscCompileList.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# This file is consumed in tools\prepublish.js
+# '#' at the start of the line is treated as a comment and is ignored.
+# whitespace empty lines are also ignores.
+#
+# This file is to be used if you do not wish to webpack a .ts file
+# If you do not want it to be webpacked:
+# 1. Please make sure it is not in the src directory
```
Comment (bobbrow) [inline Extension/tscCompileList.txt:7]: oh... I was thinking we would want to move the settings UI into the src folder at some point.  Not a big deal, I guess, but we can revisit this when we make that call.

File: Extension/package.json [hunk 4]
```diff
@@ -1472,11 +1474,13 @@
...
     "mkdirp": "~0.5.1",
     "tmp": "~0.0.33",
-    "vscode-cpptools": "2.1.1",
+    "vscode-cpptools": "^2.1.2",
     "vscode-debugadapter": "~1.33.0",
     "vscode-debugprotocol": "~1.33.0",
     "vscode-extension-telemetry": "^0.1.1",
     "vscode-languageclient": "5.1.1",
+    "webpack": "^4.29.6",
+    "webpack-cli": "^3.2.3",
```
Comment (bobbrow) [inline Extension/package.json:1483]: Do these additions generate a new 3rd party licenses file that we need to check in?

**Review by bobbrow — APPROVED** [review]

File: Extension/package.json (hunk 4 above)
Comment (WardenGnaw) [reply Extension/package.json:1483]: These should actually be under devDependencies. I don't think we have to ship any webpack code.

File: Extension/tscCompileList.txt (hunk 3 above)
Comment (WardenGnaw) [reply Extension/tscCompileList.txt:7]: IMO and with the addition of UI, we should move towards more of a [webpage](https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/Dealing_with_files) styled folder structure. 

E.g.
```
dist/
   main.js
   *.html
   images/
   styles/
   scripts/
```

File: Extension/package.json (hunk 4 above)
Comment (sean-mcmanus) [reply Extension/package.json:1483]: Oops, I didn't see that this change wasn't made. I'll submit a fix.

**Comment by philkuz** [comment]
Linking #3213 as it is fixed with this merge.

**Pull request merged**